src/mon/PGMap.cc

```diff
@@ -539,6 +539,13 @@ void PGMap::stat_pg_update(const pg_t pgid, pg_stat_t& s,
     s.blocked_by == n.blocked_by;
 
   stat_pg_sub(pgid, s, sameosds);
+
+  // if acting_primary has shift to an just restored osd, and pg yet to finish
+  // peering, many attributes in current stats remain stale. others seem don't
+  // mater much while faulty last_active will make "pg stuck in" check unhappy.
+  if ((s.acting_primary != n.acting_primary) &&
+      !(n.state & (PG_STATE_ACTIVE | PG_STATE_PEERED)))
+    n.last_active = s.last_active;
```

Comment:
@LiumxNL neither do i understand this fix. if the bug is that the just-restored OSD where PG is still peering and not active, its last_active should not be updated. see void `PG::publish_stats_to_osd()`, the last_active is set only if the new state is `PG_STATE_ACTIVE`.